### ISSUE TITLE
Site hanged if Discount products with these facets by 5% is active

### ISSUE THREAD

**Issue by hkjeffchan** [vendurehq/storefront-angular-starter#16]
The shop hanged if discount is applied with facet rules.

Tried firefox and chrome with the latest verison.

**Comment by michaelbromley** [comment]
Reproduction:

1. Create a Promotion with 5% off all products with the "photo" facetValue
2. In the demo storefront, add a camera to the order
3. Browser hangs.

Most likely an infinite change-detection loop in Angular.

**Comment by talesmgodois** [comment]
I am trying to simulate here. But Looks like it's fine. is **order** the same as **Cart**?

**Comment by michaelbromley** [comment]
@talesmgodois Yes, `Order` is the name of the entity used by the server, whereas "Cart" is used in the storefront as a more familiar term for customers.
